### ISSUE TITLE
Fatal error in ../src/platform-linux.cc, line 776

### ISSUE THREAD

**Issue by znmeb** [ropensci/jsonvalidate#12]
Arch Linux, with Arch User Repository package v8-3.14.5 installed:

```
> devtools::install_github("ropensci/jsonvalidate")
Downloading GitHub repo ropensci/jsonvalidate@master
✔  checking for file ‘/tmp/Rtmpu2Ayok/remotes5832f83a615/ropensci-jsonvalidate-6ec811e/DESCRIPTION’ ...
─  preparing ‘jsonvalidate’:
✔  checking DESCRIPTION meta-information ...
─  checking for LF line-endings in source and make files and shell scripts
─  checking for empty or unneeded directories
─  building ‘jsonvalidate_1.0.0.tar.gz’
   
Installing package into ‘/home/znmeb/R/x86_64-pc-linux-gnu-library/3.5’
(as ‘lib’ is unspecified)
* installing *source* package ‘jsonvalidate’ ...
** R
** inst
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
** building package indices
** installing vignettes
** testing if installed package can be loaded
sh: line 1:  2036 Trace/breakpoint trap   (core dumped) '/usr/lib64/R/bin/R' --no-save --slave 2>&1 < '/home/TMPDIR/RtmpoTO5a2/file7db65937624'


#
# Fatal error in ../src/platform-linux.cc, line 776
# CHECK_EQ(0, result) failed
#   Expected: 0
#   Found: 22
#

ERROR: loading failed
* removing ‘/home/znmeb/R/x86_64-pc-linux-gnu-library/3.5/jsonvalidate’
Error in i.p(...) : 
  (converted from warning) installation of package ‘/tmp/Rtmpu2Ayok/file5835fbe218b/jsonvalidate_1.0.0.tar.gz’ had non-zero exit status
```

I'm not sure what's going on here - this used to install but I don't remember how long ago that was.

**Comment by behrica** [comment]
Same for me as well on arch linux by running
install.packages("jsonvalidate")

**Comment by behrica** [comment]
I tried to reinstall everything:

pacaur -S v8-3.14

devtools::install_cran("V8",force=T)
devtools::install_github("ropensci/jsonvalidate")

But no success.

I had it definitely working as well some time ago. After  I changed to R 3.5 I had to reinstall it (today), and then it failed to install.

**Comment by znmeb** [comment]
This looks like a problem with the way Arch installs v8-3.14 - it doesn't install *everything*. The long-term solution is for the R `V8` package to require an up-to-date Linux V8, rather than one that's not supported by any Linux distro *except* to compile the R package. There's an issue to do that but I don't know if anyone's claimed ownership - https://github.com/jeroen/V8/issues/39

I'll go file an issue on this in the Arch User Repository after I test it in an Ubuntu container to see if it still works there.

**Comment by daissi** [comment]
> Which raises the question - who maintains the `apt` package in Debian, Ubuntu and other Debian derivatives? What indication is there that it will still be there in newer releases - are we locking people into long-term support Linux distros because the R `V8` package depends on this?

The next Debian release [will not provide](https://tracker.debian.org/pkg/libv8-3.14) libv8-3.14. Consequently, it will not be possible to build the R V8 package.

**Comment by znmeb** [comment]
@daissi Thanks! I guess we need to escalate the issue on the V8 repository - see https://github.com/jeroen/V8/issues/46.

**Comment by richfitz** [comment]
is this fixed with the updates that Jeroen made in packaging V8?

**Comment by prosoitos** [comment]
Have the same issue with package geojsonio

**Comment by pascaloettli** [comment]
Similar to @prosoitos but on Ubuntu 18.04.2 with V8 3.14.5-8:

```
#
# Fatal error in ../src/platform-linux.cc, line 782
# CHECK_EQ(0, result) failed
#   Expected: 0
#   Found: 22
#

Trace/breakpoint trap (core dumped)
ERROR: loading failed

```

**Comment by prosoitos** [comment]
According to https://github.com/jeroen/V8/issues/45, it seems that this is not a V8 issue, but an issue with the AUR package.

See @jeroen's comment:

> Sorry but there is nothing I can do about this. You need to talk to the maintainer of the arch v8-3.14 pkg.

Beside failing the installation of some packages which depend on V8 (e.g. geojsonio or jsonvalidate), it makes other packages which also depend on V8 and which install fine, crash upon use with the same error message (e.g. https://github.com/joelgombin/concaveman).

While waiting for the AUR package to hopefully work with R again, @JanMarvin created a repo with an R compatible PKGBUILD: https://github.com/JanMarvin/v8-R.

I just tested it and it works (both to install and to run the various packages I have which rely on V8). Yeah! Thank you @JanMarvin. I hope the changes do get incorporated in the AUR version.

In any event, since this has nothing to do with jsonvalidate, this issue should probably be closed, no?

**Comment by pascaloettli** [comment]
The problem doesn't only affect Arch Linux

**Comment by prosoitos** [comment]
True. Arch derivatives such as Manjaro, but also Debian and its derivatives (e.g. Ubuntu) seem to also be affected. So it is indeed not just the AUR v8 package that has become R incompatible. But it is not a jsonvalidate issue.

**Comment by prosoitos** [comment]
Have you tried @JanMarvin's package @pascaloettli?

**Comment by pascaloettli** [comment]
@prosoitos  I am not on Arch Linux

**Comment by prosoitos** [comment]
Oh, sorry, you said you were on Ubuntu above. My bad :worried:

**Comment by cmcaine** [comment]
User report to hopefully save others some time:

I've tried installing three different versions of v8 on arch to no avail.

v8-3.14.5, v8-3.14-bin, and from JanMarvin's repo.

I recompiled the R V8 package with `install.packages("V8")` and tested for the error with `V8::v8()`.

For now, I'm just giving up on it. If I really need to use JSON I'll just use Julia or python.

**Comment by aheilbut** [comment]
I found, per https://github.com/jeroen/V8/issues/45#issuecomment-508929952  that if I just installed the node-dev rather than v8, it all worked again

**Comment by prosoitos** [comment]
JanMarvin's repo still works for me without any problem (on Arch).
